### DISCUSSION TITLE
Question on Canary deployment of InferenceService

### DISCUSSION THREAD

**Discussion by harmeet-singh-discovery** [kubeflow/kfserving#3095] [General]
/kind question

As discussed in [this](https://kserve.github.io/website/0.9/modelserving/v1beta1/rollout/canary-example/#canary-rollout-example) example of canary rollout, user needs to carry out three steps:
1. Update the InferenceService by adding the canaryTrafficPercent field to the predictor component
2. After rolling out the canary model, Run Prediction to ensure no failure happens with new deployment
3. Promote the canary model by removing the canaryTrafficPercent field and re-applying the InferenceService custom resource

If I understand correctly, this means I have to deploy InferenceService at-least twice. 

Question: Is it possible to define Canary Deployment in such a way that KServe automatically promote Canary model to receive 100% if it passes health checks ?

**Comment by yuzisun** [comment]
Canary is meant for more advanced testing such as model accuracy testing, performance testing which can’t easily be defined on inference service. If it is just health check then the default rollout already does this by moving 100% traffic from old revision to new revision once it passes the readiness probes. Though you can still build a workflow using for example argo to automate these canary steps if that makes sense.
